gitlabbot/gitlabbot/util.go

```diff
@@ -0,0 +1,178 @@
+package gitlabbot
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"github.com/keybase/go-keybase-chat-bot/kbchat/types/chat1"
+	"github.com/xanzy/go-gitlab"
+	"net/http"
+	"strings"
+
+	"github.com/keybase/go-keybase-chat-bot/kbchat"
+	"github.com/keybase/managed-bots/base"
+)
+
+func makeSecret(repo string, convID chat1.ConvIDStr, secret string) string {
```

Comment:
Seems like a bunch of these can be shared with githubbot instead of duplicated? It would be nice especially if the message formatting was shared

Reply:
message formatting would definitely be a good thing to share between these bots

Reply:
Some of the miscellaneous Util functions can be shared, I can put them in a util class under base? But I think the formatting is just different enough that making some abstraction would be more complicated than just having them separate.

Reply:
Aren't these ultimately formatting the same class of events?

Reply:
@Daniel-Wang you could maybe add a bunch of arguments to the functions for branch, repo, test name, etc. and then pass in different things from the gitlab/github event types

Reply:
Okay, yeah that's true. Can I do that in another PR, sorry this one is really big lol

Reply:
Sure